runtime-sdk/src/modules/core/mod.rs

```diff
@@ -30,10 +40,206 @@ pub enum Error {
     #[error("insufficient balance to pay fees")]
     #[sdk_error(code = 5)]
     InsufficientFeeBalance,
+
+    #[error("invalid argument")]
+    #[sdk_error(code = 6)]
+    InvalidArgument,
+
+    #[error("gas overflow")]
+    #[sdk_error(code = 7)]
+    GasOverflow,
+
+    #[error("out of gas")]
+    #[sdk_error(code = 8)]
+    OutOfGas,
+
+    #[error("batch gas overflow")]
+    #[sdk_error(code = 9)]
+    BatchGasOverflow,
+
+    #[error("batch out of gas")]
+    #[sdk_error(code = 10)]
+    BatchOutOfGas,
+}
+
+/// Parameters for the accounts module.
+#[derive(Clone, Debug, Serialize, Deserialize)]
+#[serde(deny_unknown_fields)]
+pub struct Parameters {
+    #[serde(rename = "batch_gas")]
+    pub batch_gas: u64,
+}
+
+impl Default for Parameters {
+    fn default() -> Self {
+        Self { batch_gas: 0 }
+    }
+}
+
+impl module::Parameters for Parameters {
+    type Error = std::convert::Infallible;
+}
+
+pub trait API {
+    fn use_gas(ctx: &mut TxContext<'_, '_>, gas: u64) -> Result<(), Error>;
+    fn batch_use_gas(ctx: &mut DispatchContext<'_>, gas: u64) -> Result<(), Error>;
+}
+
+/// Genesis state for the accounts module.
+#[derive(Clone, Debug, Serialize, Deserialize)]
+#[serde(deny_unknown_fields)]
+pub struct Genesis {
+    #[serde(rename = "parameters")]
+    pub parameters: Parameters,
+}
+
+impl Default for Genesis {
```

Comment:
Feel free to derive `Default`, #92 changes a bunch of these to use default derivation instead of explicitly defining these.

Reply:
ok

Reply:
converted from #92